### ISSUE TITLE
Installation documentation without root privileges

### ISSUE THREAD

**Issue by pheidippides** [marshallward/f90nml#49]
Hi,

Thanks a lot for your work, it's going to make my life way easier!

There's a little typo in the readme for installation without root privileges, you need to invert `install` and `--user`:
```python setup.py install --user```

Cheers

**Comment by marshallward** [comment]
Thanks, sorry about that.  I've updated the README and docs.

**Comment by tclune** [comment]
I get the following error when attempting to install without privileges:

```
bash-3.2$ python2 setup.py install --user
python2 setup.py install --user
running install
error: can't combine user with prefix, exec_prefix/home, or install_(plat)base
```

Works fine with pip and privileged install.   Quite possibly this is just that I cannot mix non-elevated installation in my existing python sandbox?

**Comment by marshallward** [comment]
As mentioned in the review, I'm not able to replace this, but Dr Google does point to a potential issue:

    https://stackoverflow.com/questions/4495120/combine-user-with-prefix-error-with-setup-py-install

which points to an issue with virtualenv in a OpenSUSE and Red Hat:

    https://bugs.launchpad.net/virtualenv/+bug/248558

I don't use virtualenv much, but do have a Red Hat machine available, and will try to reproduce it there.

**Comment by tclune** [comment]
Note: my system is vanilla macOS, so the underlying culprit may have been misdiagnosed in that thread.   

But the <--prefix=> mentioned there does resolvethe problem for me.   Maybe just a parenthetical note in the README.md that some users might need the extra option?

**Comment by marshallward** [comment]
Sure, happy to mention it.  Were you using virtualenv?  Or just inside a basic shell session?

**Comment by tclune** [comment]
Just bash.   No virtual anything.

**Comment by zbeekman** [comment]
On macOS (Mojave) I have no problem installing via pip and pip3 with brewed python 2.7 and 3.7. I can do this install with both `--user` and without, and from PyPi and from a local git clone. I can also install via `setup.py` without the `--user` argument for both python 2 and 3. Adding the user flag, as recommended in the readme results in:

```
$ python setup.py install --user
running install
error: can't combine user with prefix, exec_prefix/home, or install_(plat)base
$ python3 setup.py install --user
running install
error: can't combine user with prefix, exec_prefix/home, or install_(plat)base
```

I wonder if something in `setup.py` needs to be tweaked to accommodate `--user`? Or if you should just tell people to `pip install --user .` instead? I didn't try the work around, but assume it would work, as it does for Tom.

**Comment by marshallward** [comment]
`pip install . --user` works for me, and seems to be becoming the recommended method for installing Python modules.  I can switch to this if it's causing problems.

It would be nice to be able to reproduce and investigate this issue though, so I will keep this open a bit longer.

**Comment by marshallward** [comment]
I've modified the `README` to install using pip rather than `setup.py`.  For root installs, this is the command:

    pip install .

For user installs, this is the command

    pip install -e . --user

Hopefully this will work more widely than running `setup.py` directly.

I've left the legacy `setup.py` instructions, but if it's clear that they're not needed then I may remove them.  (I don't personally use `pip` much for this, so it's a bit new to me)

**Comment by marshallward** [comment]
This has more or less merged with #96 so re-closing this now.
